internal/generator/vector/output/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,133 @@
+package elasticsearch
+
+import (
+	"strings"
+
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	. "github.com/openshift/cluster-logging-operator/internal/generator"
+	"github.com/openshift/cluster-logging-operator/internal/generator/fluentd/helpers"
+	vectorhelpers "github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/security"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Elasticsearch struct {
+	Desc           string
+	ComponentID    string
+	Inputs         string
+	Index          string
+	Endpoint       string
+	RetryTag       Element
+	SecurityConfig []Element
+}
+
+func (e Elasticsearch) Name() string {
+	return "elasticsearchTemplate"
+}
+
+func (e Elasticsearch) Template() string {
+	return `{{define "` + e.Name() + `" -}}
+
+[sinks.{{.ComponentID}}]
+type = "elasticsearch"
+inputs = {{.Inputs}}
+endpoint = "{{.Endpoint}}"
+index = "{{.Index}}"
+{{- end}}
+`
+}
+
+func Conf(o logging.OutputSpec, inputs []string, secret *corev1.Secret, op Options) []Element {
+	outputs := []Element{}
+	index := "default"
+	for _, input := range inputs {
+		switch input {
+		case logging.InputNameApplication:
+			index = "app-write"
+		case logging.InputNameInfrastructure:
+			index = "infra-write"
+		case logging.InputNameAudit:
+			index = "audit-write"
+		}
```

Comment:
what about user defined inputs?

Reply:
As of now its not addressing it. I will add in next stage.

Reply:
Right now I didn't handle it. Will look into it.